libraries/androidutils/src/main/kotlin/io/element/android/libraries/androidutils/assets/AssetReader.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2025 New Vector Ltd.
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
+ * Please see LICENSE files in the repository root for full details.
+ */
+
+package io.element.android.libraries.androidutils.assets
+
+import android.content.Context
+import dev.zacsweers.metro.Inject
+import io.element.android.libraries.di.annotations.ApplicationContext
+import timber.log.Timber
+import java.util.concurrent.ConcurrentHashMap
+
+/**
+ * Read asset files.
+ */
+@Inject
+class AssetReader(
+    @ApplicationContext private val context: Context,
+) {
+    private val cache = ConcurrentHashMap<String, String?>()
+
+    /**
+     * Read an asset from resource and return a String or null in case of error.
+     *
+     * @param assetFilename Asset filename
+     * @return the content of the asset file, or null in case of error
+     */
+    fun readAssetFile(assetFilename: String): String? {
+        return cache.getOrPut(assetFilename, {
+            return try {
+                context.assets.open(assetFilename)
+                    .use { asset ->
+                        buildString {
+                            var ch = asset.read()
+                            while (ch != -1) {
+                                append(ch.toChar())
+                                ch = asset.read()
+                            }
+                        }
+                    }
```

Comment:
Why not use `String(context.assets.open(assetFileName).use { it.readBytes() }) `?

Reply:
Or probably even better `context.assets.open(assetFileName).use { it.bufferedReader().readText() }`

Reply:
Yes, thanks. This code also supports UTF-8 chars, so definitely better!